pkg/backends/server.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright © 2023 The Spray Proxy Contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+package backends
+
+import (
+	"bytes"
+	"net/http"
+	"net/http/httptest"
+)
+
+type testBackend struct {
+	server *httptest.Server
+	err    error
+}
+
+func NewTestServer() *testBackend {
```

Comment:
If this code is only intended for tests, please rename the file to `server_test.go`. Go has a built in convention where files with the suffix `_test` are only compiled when running `go test` - they are not compiled with `go build` and thus won't make it into production code.

Reply:
I also moved those helper methods in a separate package as part of https://github.com/redhat-appstudio/sprayproxy/pull/33. In my case I put them in a package called test. My point is you can't use exported methods (here `NewTestServer`) from a test file, so after renaming the `backend.go` file to `backend_test.go`, my tests started failing with `undefined: test.NewTestServer`. Here is the explanation I found on StackOverflow "Files matching *_test.go are only compiled when testing the package they are part of. If you are testing package A that uses package B, you won't have access to the _test.go code from package B."
Earlier I had another issue "no non-test Go files in <directory>", but that was easy to overcome by adding a doc.go file.
Hope that saves you some time investigating.

Reply:
This file has the function to create the test server. So IF we rename it with `server_test.go` other test files using this function error out with import package. As the content inside this file has some common function, I moved it to single file.
Does moving this file under `/pkg/test` or `pkg/utils` folder would make sense?

Reply:
I prefer top level test directory and that's what I did in my PR. We already discussed using `utils` and similar names in another PR and IIRC the general sentiment was to avoid them.